velox/exec/HashTable.cpp

```diff
@@ -669,9 +670,15 @@ void HashTable<ignoreNullKeys>::checkSize(int32_t numNew) {
     // then the table lookup will become slow. Given that, we treat tombstone
     // slot as non-empty slot here to decide whether to trigger rehash or not.
   } else if (newNumDistincts > rehashSize()) {
-    // NOTE: we need to plus one here as newNumDistincts itself could be power
-    // of two.
-    auto newSize = bits::nextPowerOfTwo(newNumDistincts + 1);
+    // Now we need to decide if we want to increase capacity to the next power
```

Comment:
That's good thought. But if number of tombstones are small, we don't need to worry about the next power of two calculation and it will work pretty much the same way as the case without tombstones? Also rehash will clear the tombstones.